crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -68,8 +68,8 @@ impl C8yEndPoint {
         c8y_profile: Option<&str>,
     ) -> Result<Self, C8yEndPointConfigError> {
         let c8y_config = tedge_config.c8y.try_get(c8y_profile)?;
-        let c8y_host = c8y_config.proxy.client.host.to_string();
-        let c8y_mqtt_host = c8y_host.clone();
+        let c8y_host = c8y_config.http.or_config_not_set()?.to_string();
+        let c8y_mqtt_host = c8y_config.mqtt.or_config_not_set()?.to_string();
```

Comment:
I will revert that, as this is unrelated to this PR and makes a system working.

These values are only used for error messages, but setting the proper value here (i.e. the tenant host and not the proxy host) forces the child devices to properly configure c8y.url which is useless (this is why test of `tedge upload c8y` from a child device fails).